ts/private/ts_project.bzl

```diff
@@ -125,24 +125,28 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
     # Add user specified arguments *before* rule supplied arguments
     common_args.extend(ctx.attr.args)
 
-    outdir = _lib.join(
-        ctx.label.workspace_root,
-        _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package,
-    )
+    if (ctx.attr.out_dir and ctx.attr.out_dir != ".") or ctx.attr.root_dir:
+        # TODO: add validation that excludes is non-empty in this case, as passing the --outDir or --declarationDir flag
+        # to TypeScript causes it to set a default for excludes such that it won't find our sources that were copied-to-bin.
```

Comment:
Can we put a reference to https://github.com/microsoft/TypeScript/issues/59036 and the rules_js issue about it?

Reply:
done